### PR TITLE
feat(context-client): generalize implementation

### PR DESCRIPTION
**Pull request by petarjuki7**
### Part of refactoring of the contract clients

This does not compile yet

### CODE REVIEW

File: crates/context/config/src/client/protocol/near.rs [hunk 1]
```diff
@@ -26,7 +24,10 @@ use serde::{Deserialize, Serialize};
 use thiserror::Error;
 use url::Url;
 
-use super::{Operation, Transport, TransportRequest};
+use super::{private, Near};
```
Comment (miraclx) [inline crates/context/config/src/client/protocol/near.rs]: `Near` should be in this module instead

File: crates/context/config/src/client/protocol/near.rs [hunk 2]
```diff
@@ -26,7 +24,10 @@ use serde::{Deserialize, Serialize};
...
+use crate::client::{Operation, Transport, TransportRequest};
+
+impl private::Protocol for Near {}
```
Comment (miraclx) [inline crates/context/config/src/client/protocol/near.rs]: `Protocol` no longer needs to be private, we can remove that

File: crates/context/src/lib.rs [hunk 3]
```diff
@@ -73,7 +73,7 @@ impl ContextManager {
         network_client: NetworkClient,
     ) -> EyreResult<Self> {
         let client_config = config.client.clone();
-        let config_client = ContextConfigClient::from_config(&client_config);
+        let config_client = Client::from_config(&client_config);
```
Comment (miraclx) [inline crates/context/src/lib.rs]: alias this to `ExternalClient` or something like that

from my reorganization comment, this should later be in a module called

`calimero_external::Client`

File: crates/context/config/src/client/env/config.rs [hunk 4]
```diff
@@ -1,7 +1,19 @@
+use std::collections::BTreeMap;
+
+use query::appilcation_revision::{ApplicationRevision, Revision};
```
Comment (chefsale) [inline crates/context/config/src/client/env/config.rs]: Appilcation?

File: crates/context/config/src/client/env/config/mutate/context.rs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    pub(crate) kind: RequestKind<'a>,
+}
+
+impl<'a> Method<Mutate<'a>> for Near {
+    const METHOD: &'static str = "mutate";
+
+    type Returns = ();
+
+    fn encode(params: &Mutate) -> eyre::Result<Vec<u8>> {
+        let signed = Signed::new(&Request::new(todo!(), params.kind), |b| {
```
Comment (petarjuki7) [inline crates/context/config/src/client/env/config/mutate/context.rs]: Was not sure what is the signer_id and what is the secret_key, tried to figure out from what parameters are passed to it from src/lib.rs but coudln't figure it out.
Help appreciated.
cc: @miraclx

File: crates/context/config/src/client/env/mod.rs [hunk 6]
```diff
@@ -0,0 +1,2 @@
+pub mod config;
+// pub mod proxy;
```
Comment (xilosada) [inline crates/context/config/src/client/env/mod.rs]: I removed this to reduce the scope of the PR, `crates/context/config/src/client/env/proxy.rs` should be addressed independently

File: crates/context/config/src/client/env/proxy.rs [hunk 7]
```diff
@@ -0,0 +1,101 @@
+use crate::client::protocol::near::Near;
```
Comment (xilosada) [inline crates/context/config/src/client/env/proxy.rs]: I left the file just for future reference

**Comment by petarjuki7** [comment]
Is this fine to be merged now?
cc: @miraclx  @xilosada

**Review by miraclx — APPROVED** [review]

**Pull request merged**